File: core/src/main/java/com/jinloes/claudereviews/model/ReviewResult.java
package com.jinloes.claudereviews.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ReviewResult {
    private String summary;
    private List<LineComment> lineComments;
    private String verdict; // "APPROVE", "REQUEST_CHANGES", "COMMENT"

    public ReviewResult() {}

    public ReviewResult(String summary, String verdict, List<LineComment> lineComments) {
        this.summary = summary;
        this.verdict = verdict;
        this.lineComments =
                lineComments != null
                        ? new java.util.ArrayList<>(lineComments)
                        : new java.util.ArrayList<>();
    }

    public String getSummary() {
        return summary != null ? summary : "";
    }

    public List<LineComment> getLineComments() {
        if (lineComments == null) lineComments = new java.util.ArrayList<>();
        return lineComments;
    }

    public String getVerdict() {
        return verdict != null ? verdict : "COMMENT";
    }
}

File: core/src/commonMain/kotlin/com/jinloes/prpilot/model/ReviewResult.kt
package com.jinloes.prpilot.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Holds the AI-generated review: a markdown summary, a verdict, and a list of inline line
 * comments. Default parameter values allow both no-arg Jackson deserialization and
 * kotlinx.serialization decoding.
 */
@Serializable
class ReviewResult() {

    // Backing fields use distinct names to avoid JVM signature clashes with accessor methods.
    @SerialName("summary") @Suppress("PropertyName") internal var _summary: String? = null
    @SerialName("verdict") @Suppress("PropertyName") internal var _verdict: String? = null
    @SerialName("lineComments") @Suppress("PropertyName") internal var _lineComments: MutableList<LineComment>? = null

    constructor(
        summary: String,
        verdict: String,
        lineComments: List<LineComment>?,
    ) : this() {
        _summary = summary
        _verdict = verdict
        _lineComments = lineComments?.toMutableList() ?: mutableListOf()
    }

    fun getSummary(): String = _summary ?: ""
    fun setSummary(value: String) { _summary = value }

    fun getVerdict(): String = _verdict ?: "COMMENT"
    fun setVerdict(value: String) { _verdict = value }

    fun getLineComments(): MutableList<LineComment> {
        if (_lineComments == null) _lineComments = mutableListOf()
        return _lineComments!!
    }
    fun setLineComments(value: MutableList<LineComment>) { _lineComments = value }
}
